src/main/java/com/commercetools/emailprocessor/model/TenantConfiguration.java

```diff
@@ -100,6 +111,10 @@ boolean isValid() {
             LOG.error(String.format(errorMessage, "endpointUrl"));
             isValid = false;
         }
+        if (StringUtils.isEmpty(encryptionKey)) {
+            LOG.error(String.format(errorMessage, "encryptionKey"));
```

Comment:
I think it is not very good design when you log in a method that is called `isValid` because it is kind of a side-effect. So as a consumer of this method, I expect to only return if it's valid or not (`true` or `false`) but not logging as well. The consumer is expected to be the one doing the logging.

Reply:
4b1eaf7